NorthstarDedicatedTest/AntiRCE.cpp

```diff
@@ -0,0 +1,160 @@
+#include "pch.h"
+#include "AntiRCE.h"
+#include "cvar.h"
+
+static bool initialized = false;
+
+void OnGameFileAccess(const char* pathArg, bool readOnly)
+{
+	if (!initialized)
+		return;
+	static char curDirectory[MAX_PATH];
+	static DWORD curDirectoryResult = GetCurrentDirectoryA(MAX_PATH, curDirectory);
+	static int curDirectoryLen = strlen(curDirectory);
+
+	static ConVar* ns_log_all_file_access = new ConVar(
+		"ns_log_all_file_access",
+#ifdef _DEBUG
+		"1",
+#else
+		"0",
+#endif
+		FCVAR_GAMEDLL,
+		"Log all file access from the game");
+
+	if (!pathArg)
+		return;
+
+	// Resolve full path
+	char path[MAX_PATH] = {};
+	_fullpath(path, pathArg, MAX_PATH);
+
+	if (ns_log_all_file_access->GetBool())
+	{
+		spdlog::info("File handle opened to \"{}\", thread ID: {}", path, (void*)GetCurrentThreadId());
+	}
+
+	int pathLen = strlen(path);
+
+	const char* extension = strrchr(path, '.');
+	if (extension)
+	{
+		if (!readOnly)
+		{
+			constexpr const char* EVIL_EXTENTIONS[] = {".dll", ".exe", ".bat", ".cmd", ".com"};
```

Comment:
add .ps1, .vbs, .vb, possibly .js and .sys, .msi, .scr, .lnk, .cpl

Reply:
Adding in new commits, thx

Reply:
Added in new commit, thanks!